Prepare docs for 3.5 (#20852)
matplotlib/matplotlib: doc/users/prev_whats_new/whats_new_3.5.0.rst

==============================
What's new in Matplotlib 3.5.0
==============================

For a list of all of the issues and pull requests since the last revision, see
the :ref:`github-stats`.

.. contents:: Table of Contents
   :depth: 4

.. toctree::
   :maxdepth: 4

Figure and Axes creation / management
=====================================

Figure now has ``draw_no_output`` method
----------------------------------------

> Rarely, the user will want to trigger a draw without making output to either
> the screen or a file. This is useful for determining the final position of
> artists on the figure that require a draw, like text artists. This could be
> accomplished via ``fig.canvas.draw()`` but has side effects, sometimes requires
> an open file, and is documented on an object most users do not need to access.
> The `.Figure.draw_no_output` is provided to trigger a draw without pushing to
> the final output, and with fewer side effects.

Reviewer: This paragraph is very hard to follow if you don't already know what draw means, so I'm wondering if it's linking to some general documentation about draw? Or if there's a one sentence summary of draw? Or using the classic example of `ax.get_labels()` returns blank before draw?

```suggestion
Rarely, the user may want to trigger a draw without rendering output to either
the screen or a file. This is useful for determining the position of
artists that require a draw, like text artists. This could be
accomplished via ``fig.canvas.draw()`` but has side effects, sometimes requires
an open file, and is documented on an object most users do not need to access.
The `.Figure.draw_no_output` is provided to trigger a draw without rendering and with fewer side effects.
```

Colormaps can be obtained using item access::

    import matplotlib.pyplot as plt
    cmap = plt.colormaps['viridis']

To register new colormaps use::

    plt.colormaps.register(my_colormap)

We recommend to use the new API instead of the `~.cm.get_cmap` and
`~.cm.register_cmap` functions for new code. `matplotlib.cm.get_cmap` and
`matplotlib.cm.register_cmap` will eventually be deprecated and removed.
Within `.pyplot`, ``plt.get_cmap()`` and ``plt.register_cmap()`` will continue
to be supported for backward compatibility.

Image interpolation now possible at RGBA stage
----------------------------------------------

Images in Matplotlib created via `~.axes.Axes.imshow` are resampled to match
the resolution of the current canvas. It is useful to apply an anto-aliasing
filter when downsampling to reduce Moiré effects. By default, interpolation is
done on the data, a norm applied, and then the colormapping performed.

However, it is often desireable for the anti-aliasing interpolation to happen
in RGBA space, where the colors are interpolated rather than the data. This
usually leads to colors outside the colormap, but visually blends adjacent

> colors, and is what browsers and other image processing software does.

Reviewer 2: ```suggestion
colors, and is what browsers and other image processing software do.
```
